### PR TITLE
[now dev] Bundle `canary` builders for Now CLI canary

### LINKED ISSUE #2641: Ship canary Builders with canary Now CLI
Currently, we're shipping stable Builders with stable and canary Now CLI.

To make it easier to test them, we should ship canary Builders with canary Now CLI. In turn, we also need to make now/build-utils aware of this (that's where we select the default builders for zero config).

### PR DESCRIPTION
**Pull request by TooTallNate**
Closes #2641.

<strike>I'm still adding unit tests for the "filter" function since the logic is not very straightforward and there's many edge cases.</strike>

### CODE REVIEW

**Review by AndyBitz — COMMENTED** [review]
The code looks good, but I'm wondering if it would make sense to include both versions for Now CLI canary, since I don't think that just because someone uses Now CLI canary they'd like to use canary builders as well

File: scripts/build.js [hunk 1]
```diff
@@ -16,10 +16,14 @@ const pkg = require('../package.json');
...
 const dirRoot = join(__dirname, '..');
 
+const bundledBuilders = Object.keys(pkg.devDependencies)
+  .filter(d => d.startsWith('@now/'));
+
 async function createBuildersTarball() {
-  const builders = Object.keys(pkg.devDependencies)
-    .filter(d => d.startsWith('@now/'))
-    .map(d => `${d}@${pkg.devDependencies[d]}`);
+  const distTag = pkg.version.includes('-')
```
Comment (styfle) [inline scripts/build.js]: I think you should use `npa()` here instead of includes/match. I already wrote a check to use canary/stable. See the function `getNpmVersion()` in `builder-cache.ts`.

Comment (TooTallNate) [reply scripts/build.js]: I think `semver.parse()` is actually more appropriate in this case. Updating.

**Comment by TooTallNate** [comment]
@AndyBitz I'd rather not, as there's not really a "clean" way to accomplish that without including _two_ tarballs (one for stable and one for canary) and would make the logic more complicated.

Users can still use stable builders, the `installPackages()` function blocks to ensure correctness before booting up the `now dev` server.

**Review by AndyBitz — APPROVED** [review]

File: scripts/build.js [hunk 2]
```diff
@@ -16,10 +17,20 @@ const pkg = require('../package.json');
 
 const dirRoot = join(__dirname, '..');
 
+const bundledBuilders = Object.keys(pkg.devDependencies)
+  .filter(d => d.startsWith('@now/'));
+
+function getDistTag(version) {
```
Comment (styfle) [inline scripts/build.js]: Looks like you defined this function twice.

Can it live in one place to keep it DRY?

File: src/util/dev/builder-cache.ts [hunk 3]
```diff
@@ -172,6 +183,39 @@ export function getBuildUtils(packages: string[]): string {
...
+    if (parsedInstalled.type !== 'version') {
+      return true;
+    }
+    const semverInstalled = semver.parse(parsedInstalled.rawSpec);
+    if (!semverInstalled) {
+      return true;
+    }
+    if (semverInstalled.prerelease.length > 0) {
+      return semverInstalled.prerelease[0] !== distTag;
+    } if (distTag === 'latest') {
```
Comment (styfle) [inline src/util/dev/builder-cache.ts]: Should this be `else if` or on a new line?

Comment (styfle) [reply src/util/dev/builder-cache.ts]: You could probably remove the if statement and use `return distTag !== 'latest`

File: scripts/build.js (hunk 2 above)
Comment (TooTallNate) [reply scripts/build.js]: I had to convert the build script to TypeScript, but ya, done in 6e6b0bb3.

**Review by styfle — APPROVED** [review]
